IdentifierTests/KIFUITestActor-IdentifierTests.m

```diff
@@ -158,25 +159,28 @@ - (void)setValue:(float)value forSliderWithAccessibilityIdentifier:(NSString *)a
 - (void)waitForFirstResponderWithAccessibilityIdentifier:(NSString *)accessibilityIdentifier
 {
 	[self runBlock:^KIFTestStepResult(NSError **error) {
-		UIResponder *firstResponder = [[[UIApplication sharedApplication] keyWindow] firstResponder];
-		if ([firstResponder isKindOfClass:NSClassFromString(@"UISearchBarTextField")]) {
-			do {
-				firstResponder = [(UIView *)firstResponder superview];
-			} while (firstResponder && ![firstResponder isKindOfClass:[UISearchBar class]]);
-		}
-		UIResponder<UIAccessibilityIdentification>* firstResponderIdentification = nil;
-		if ([firstResponder conformsToProtocol:@protocol(UIAccessibilityIdentification)])
-		{
-			firstResponderIdentification = (UIResponder<UIAccessibilityIdentification>*)firstResponder;
-		}
-		else
-		{
-			[self failWithError:[NSError KIFErrorWithFormat:@"First responder does not conform to UIAccessibilityIdentification %@",  NSStringFromClass([firstResponder class])] stopTest:YES];
-			
-		}
-		KIFTestWaitCondition([[firstResponderIdentification accessibilityIdentifier] isEqualToString:accessibilityIdentifier],
+        BOOL didMatch = NO;
+        NSString *foundIdentifier = nil;
+        for (UIResponder __strong *firstResponder in [[UIApplication sharedApplication] firstResponders]) {
+            if ([firstResponder isKindOfClass:NSClassFromString(@"UISearchBarTextField")]) {
+                do {
+                    firstResponder = [(UIView *)firstResponder superview];
+                } while (firstResponder && ![firstResponder isKindOfClass:[UISearchBar class]]);
+            }
+
+            if ([firstResponder conformsToProtocol:@protocol(UIAccessibilityIdentification)]) {
+                foundIdentifier = [(UIResponder<UIAccessibilityIdentification> *)firstResponder accessibilityIdentifier];
+            }
+
+            if ([foundIdentifier isEqualToString:accessibilityIdentifier]) {
+                didMatch = YES;
+                break;
+            }
+        }
+
+		KIFTestWaitCondition(didMatch,
```

Comment:
NIT: indentation seems off

Reply:
Hmm looks like there's a mix of spaces & tabs in this file, so it looks fine with my config but wrong for github's. I'll fix all in file to convert them all to spaces. It'll dirty up the commit so I'd advise ignoring whitespace afterward.

Reply:
Thanks for taking that up!